src/plugins/lvm.c

```diff
@@ -1988,8 +1988,22 @@ BDLVMLVdata** bd_lvm_lvs (const gchar *vg_name, GError **error) {
         if (table && (num_items == 16)) {
             /* valid line, try to parse and record it */
             lvdata = get_lv_data_from_table (table, TRUE);
-            if (lvdata)
-                g_ptr_array_add (lvs, lvdata);
+            if (lvdata) {
+                /* ignore duplicate entries in lvs output, these are caused by multi segments LVs */
+                for (gsize i = 0; i < lvs->len; i++) {
+                    if (g_strcmp0 (((BDLVMLVdata *) g_ptr_array_index (lvs, i))->lv_name, lvdata->lv_name) == 0) {
```

Comment:
There's a convenient `g_ptr_array_find_with_equal_func()` function for searching through the array but it wouldn't help much to relieve the clutter I guess.

Reply:
It was added in 2.54 and I didn't want to bump glib required version.